pkg/endpoint/endpoint.go

```diff
@@ -283,10 +255,24 @@ func (e *EndpointStatus) getAndIncIdx() int {
 	if e.Index >= maxLogs {
 		e.Index = 0
 	}
+	if e.Index < len(e.Log) &&
+		e.Log[e.Index] == e.CurrentStatus &&
+		e.CurrentStatus.Status.Code != OK {
+		e.Index++
+	}
 	return idx
 }
 
-func (e *EndpointStatus) addStatusLog(s *statusLog) {
+func (e *EndpointStatus) addStatusLog(s *statusLogMsg) {
+	if e.CurrentStatus.Status.Code == OK {
```

Comment:
Can you explain this priority system a bit. I assume the point is to not undo failure state unless we know that the actual failure is gone. How does this priority actual fix it?

Don't we need a bitmask which indicates the status of each failure mode and if any of them is set to failure the entire endpoint is marked as failed?